notebooks/ima_viewer/ima_viewer.ipynb

```diff
@@ -0,0 +1,587 @@
+    "- *astroquery.mast Observations* for downloading data from MAST\n",
+    "- *imageio* to make and save animated GIFs\n",
+    "- *ipympl* as a matplotlib backend to enable interactive widgets\n",
+    "- *IPython.display* to display widgets and other components\n",
+    "- *ipywidgets* to make widgets for plotting\n",
+    "- *matplotlib pyplot* for plotting data\n",
+    "- *matplotlib.ticker* to configure tick locating and formatting \n",
+    "- *matplotlib.colors* to normalize images\n",
+    "- *numpy* for calculating limit percentiles\n",
+    "- *os* for verifying and creating paths\n",
+    "- *shutil* for moving files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "2e1bf798",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy.io import fits\n",
+    "from astroquery.mast import Observations\n",
+    "import imageio.v2 as imageio\n",
+    "from IPython.display import display\n",
+    "from ipywidgets import interact, interactive, Layout\n",
+    "import ipywidgets as widgets\n",
+    "import matplotlib\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.ticker as tick\n",
+    "from matplotlib.colors import Normalize\n",
+    "import numpy as np\n",
+    "import os\n",
+    "import shutil\n",
+    "%matplotlib ipympl\n",
+    "\n",
+    "from ima_tools import make_interactive_plot, gif_settings_widgets, make_frames, make_gif"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "61c78a3e",
+   "metadata": {},
+   "source": [
+    "<a id=\"display\"></a>\n",
+    "## 2. Display IMA files \n",
+    "\n",
+    "In this section, we will download an IMA file (if needed), create an interactive plot of the IMA file, and then save the output as frames and a composite animated GIF. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "83cadf65",
+   "metadata": {},
+   "source": [
+    "<a id=\"data\"></a>\n",
+    "### 2.1 Data\n",
+    "\n",
+    "If you already have an IMA file you'd like to visualize, you can define the path to the file in the cell below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "be885dc7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#filepath = path/to/file_ima.fits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "63f6a714",
+   "metadata": {},
+   "source": [
+    "If you've defined a path in the cell above, you can now skip to the next subsection. \n",
+    "\n",
+    "Otherwise, continue on to download a file from MAST."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72ca38b4",
+   "metadata": {},
+   "source": [
+    "Let's download the IMA for a particular observation, which has a rootname or observation ID of `idpq32w0q`. \n",
+    "\n",
+    "Importantly, even if we know the observation ID, if the particular observation is part of an association, the `query_criteria` function will not be able to find the exact observation.  "
```

Comment:
I like that you included this step since that's a very counterintuitive part of astroquery